src/app/targetDownloader.js

```diff
@@ -690,32 +690,36 @@ createNameSpace("realityEditor.app.targetDownloader");
 
     }
 
+ // let schema = {
+ //        "type": "object",
+ //        "items": {
+ //            "properties": {
+ //                "obj": {"type": "string", "minLength": 1, "maxLength": 50, "pattern": "^[A-Za-z0-9_]*$"},
+ //                "server" : {"type": "string", "minLength": 0, "maxLength": 2000, "pattern": "^[A-Za-z0-9~!@$%^&*()-_=+|;:,.]"},
+ //            },
+ //            "required": ["server", "obj"],
+ //            "expected": ["server", "obj"],
+ //        }
+ //    }
+
+    const schema = new ToolSocket.Schema([
+        new ToolSocket.Schema.StringValidator('obj', {minLength: 1, maxLength: 50, pattern: /^[A-Za-z0-9_]*$/, required: true, expected: true}),
+        new ToolSocket.Schema.StringValidator('server', {minLength: 0, maxLength: 2000, pattern: /^[A-Za-z0-9~!@$%^&*()-_=+|;:,.]/, required: true, expected: true})
+    ])
+
     /**
      * Uses a combination of IP address and object name to locate the ID.
      * e.g. "http(s)://10.10.10.108:8080/obj/monitorScreen/target/target.xml" -> ("10.10.10.108", "monitorScreen") -> object named monitor screen with that IP
      * @param {string} fileName
      */
-
- let schema = {
-        "type": "object",
-        "items": {
-            "properties": {
-                "obj": {"type": "string", "minLength": 1, "maxLength": 50, "pattern": "^[A-Za-z0-9_]*$"},
-                "server" : {"type": "string", "minLength": 0, "maxLength": 2000, "pattern": "^[A-Za-z0-9~!@$%^&*()-_=+|;:,.]"},
-            },
-            "required": ["server", "obj"],
-            "expected": ["server", "obj"],
-        }
-    }
-
     function getObjectIDFromFilename(fileName) {
-        let urlObj = io.parseUrl(fileName, schema)
-        if (!urlObj) {
-            console.warn('io.parseUrl failed. this may cause targets not to download', fileName);
+        let parsedUrl = schema.parseUrl(new URL(fileName));
```

Comment:
This may apply to other uses of this API but this is what just occured to me: this could throw if `fileName` is not a URL which would potentially change behavior